modules/local/bcftools_annotate/main.nf

```diff
@@ -0,0 +1,42 @@
+process BCFTOOLS_ANNOTATE {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::bcftools=1.14" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bcftools:1.14--h88f3f91_0':
+        'quay.io/biocontainers/bcftools:1.14--h88f3f91_0' }"
+
+    input:
+    tuple val(meta), path(input)
+
+    output:
+    tuple val(meta), path("*.vcf.gz"), optional:true , emit: vcf
+    tuple val(meta), path("*.bcf")   , optional:true , emit: bcf
+    path "versions.yml"              , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    def matcher = input =~ /vcf/
+    def output_suffix = matcher ? "vcf.gz" : "bcf"
+    def output_type_compressed = matcher ? "z" : "b"
```

Comment:
So this will output a bcf unless the input is a vcf.gz?

Reply:
(discussed this on huddle but in case anyone else is reading) if substring "vcf" is matched, `output_suffix="vcf.gz"` and `output_type_compressed="z"` otherwise bcf

Reply:
aha 😄